### ISSUE TITLE
Clarify rule for setting facet types

### ISSUE THREAD

**Issue by bobylito** [algolia/algoliasearch-helper-js#436]
Currently, a facet can be set to multiple type. But it doesn't work properly due to the implementation of disjunctive and hierarchical facets.

Proposition: enforce a single type, and document it.

**Comment by vvo** [comment]
Do you mean by enforce => Throw?

**Comment by bobylito** [comment]
Yes I would say. But there is gradiant of options like not letting the user do it but do not break anything (that's a silly one I guess)
